src/library/features/crates/cratetreemodel.cpp

```diff
@@ -0,0 +1,60 @@
+#include <QVector>
+
+#include "library/trackcollection.h"
+#include "library/features/crates/cratetreemodel.h"
+
+#include "util/db/dbid.h"
+
+namespace {
+
+    const CrateId kRootId(CrateId(-1));
```

Comment:
const CrateId kRootId;

But this constant is of limited use anyway, because you implicitly use the assumption kRootId == CrateId() all over in your the code. There is also another assumption: Crate() stands for the root crate. I'm still not convinced if this is appropriate or if we better should use a nullptr. The crate object contains is defined by a collection of different attributes and only the default constructed CrateId() qualifies it to be a root crate.